### ISSUE TITLE
v0.8.0 Release Plan

### ISSUE THREAD

**Issue by chchwy** [pencil2d/pencil#1891]
Time to prepare for the next 0.8.0 release.

Please carefully review the milestone: 
→→→ [0.8.0 milestone](https://github.com/pencil2d/pencil/milestone/3)

#### Source code

- [ ] Make sure all unit tests are working and are OK
- [ ] Test all features known to work and make sure they behave the same
- [ ] Update contributor list
- [ ] Update [MetaInfo](https://github.com/pencil2d/pencil/blob/master/app/data/org.pencil2d.Pencil2D.metainfo.xml)
  - [ ] Add entry for new release
    → Update TODOs in #1845 then merge
- [ ] Make sure 0 issues in milestone
- [ ] Add release tag to the correct commit
- [ ] Created `release/0.8.0` branch

#### Deployment
- [ ] When renaming the AppImage, also update the plaintext metadata block at the beginning of the *.AppImage.zsync file accordingly
- [ ] Build and upload manually compiled versions:
  - [ ] 32-bit Linux AppImage
- [ ] Add downloads to the [Bitbucket mirror](https://bitbucket.org/chchwy/pencil2d)
- [ ] Update download links:
  - [ ] [Pencil2d.org](https://www.pencil2d.org/download/)
  - [ ] [Github README](https://github.com/pencil2d/pencil)
- [ ] Update packages
  - [ ] [Arch Linux](https://gitlab.archlinux.org/archlinux/packaging/packages/pencil2d) @City-busz
  - [ ] [Homebrew Cask](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/p/pencil2d.rb) (macOS) @scribblemaniac
  - [ ] [Apt (Debian/Ubuntu)](https://salsa.debian.org/debian/pencil2d) @fabiomt
  - [ ] [Flatpak](https://github.com/flathub/org.pencil2d.Pencil2D) (Linux) @J5lx @eszlari
  - [ ] [RPM (Fedora)](https://src.fedoraproject.org/rpms/Pencil2D) @musicinmybrain
  - [ ] [FreeBSD](https://cgit.freebsd.org/ports/tree/graphics/pencil2d) @yurivict
  - [ ] [KCP](https://github.com/KaOS-Community-Packages/pencil2d) (KaOS) @scribblemaniac

#### Social media
Announce with release notes (or a link to the release notes) on the following platforms:
- [ ] [Pencil2d.org News](https://www.pencil2d.org/news.html)
- [ ] [Facebook](https://www.facebook.com/groups/709024775972087/)
- [ ] [Discord](https://discord.gg/8FxdV2g)
- [ ] [Tumblr](https://pencil2d-software.tumblr.com/)
- [ ] [X](https://x.com/pencil2d)
- [ ] [Transifex](https://www.transifex.com/pencil2d/pencil2d/announcements/)
- [ ] [Youtube?](https://www.youtube.com/channel/UCexMjl4rq8OyzAc7C03uKCQ)
- [ ] IRC: Update status message at Libera Chat #pencil2d @J5lx

**Comment by chchwy** [comment]
I'd like to begin preparing the v0.7.1 release since I have some availability to work on Pencil2D lately.

- I will start a quick QA session by animating daily with the `release/0.7.1` branch to verify the build stability
- I will arrange a v0.7.1 RC in a few days

Please let me know if you have any questions.

**Comment by chchwy** [comment]
There is only one issue (#1873) left in the milestone and it is nearly ready to be merged.

I will create the RC release after that.

**Comment by chchwy** [comment]
0.7.1-rc1 is available at:
https://github.com/pencil2d/pencil/releases/tag/v0.7.1-rc1

**Comment by Jose-Moreno** [comment]
@chchwy Awesome, I'll be able to fully trial the RC on the weekend. For public testing purposes, I'll post notice as soon as I have an understanding of current changes / updates compared to 0.7.0 so I can help provide support if needed.

As always, thanks everyone for the hard work!

**Comment by chchwy** [comment]
Thanks @Jose-Moreno 

Please see the following changelog, I just summarized from the git history:

## Improvements 

- #1916 Improved toolbox layout. Tools now intelligently arrange in rows with consistent spacing and can shrink to single column.

## Fixes

- #1873 Fixed frame not being drawn on the timeline 
- #1878 Fixed broken image import camera transform 
- #1881 Fixed keyframes moving beyond layer name boundaries in Timeline
- #1896 Fixed memory leak in ClipboardManager where cloned frames were not deleted
- #1896 Fixed frame data loss when copying/pasting bitmap keyframes by properly cloning backing files and implementing temporary working directory for PCL projects
- #1896 Fixed frame filename exceeding path length limit
- #1897 Fixed view rotation affecting various view handles
- #1918 Fixed SoundManager calculating frame length incorrectly

## Changes

- #1907 Disabled Timeline double-click keyframe creation to prevent accidental keyframes

## DevOps

- #1895 Added more details to FileManager error reports
- #1902 Fixed Linux build failure by running actions outside container
- #1912 Ensured QDebug is included for all qDebug uses

**Comment by Jose-Moreno** [comment]
@chchwy Thanks! I shared a simple post on socials asking for assistance with testing so we can hopefully obtain additional feedback. 
On the other hand I did a preliminary test drive for the RC. In general it's working OK under Win 11 Home 24h2. While I did run into some issues that I'm already documenting, I noticed a couple of them might be related to some of the canvas update modifications that were implemented since 0.7.0. For now I need to test thoroughly to figure out reproduction steps and see which ones are either actual bugs, regressions or simple oversights. I'll let you guys know as soon as I have more useful info.

**Comment by chchwy** [comment]
I am working on the macOS notarization https://github.com/pencil2d/pencil/issues/1922
hopefully I can get it done in a couple of days.

**Comment by eugenialoli** [comment]
Any news when the final version of 0.7.1 is going to be released?

**Comment by scribblemaniac** [comment]
@eugenialoli No. Testing of the first release candidate revealed several issues that I think should be fixed before the final release. Development is very slow right now, so there's really no telling how long it will take.

**Comment by fabiomt** [comment]
Hi,

@chchwy 
Debian package for the 0.7.1 version just uploaded to the Debian unstable repository.
In about a week should complete it's migration to testing.
This upload should be available in Ubuntu as well shortly.

**Comment by chchwy** [comment]
Thank you @fabiomt 
However, I just found that I forgot to update the AppImage by merging the PR below:
https://github.com/pencil2d/pencil2d.github.io/pull/39

Probably I will have to make a hotfix release

**Comment by musicinmybrain** [comment]
I’m shipping 0.7.1 in Fedora (all branches) and EPEL10. I’ll update again to 0.7.2 when it is ready.

**Comment by musicinmybrain** [comment]
> I’m shipping 0.7.1 in Fedora (all branches) and EPEL10. I’ll update again to 0.7.2 when it is ready.

Done.

**Comment by chchwy** [comment]
> > I’m shipping 0.7.1 in Fedora (all branches) and EPEL10. I’ll update again to 0.7.2 when it is ready.
> 
> Done.

Thank you @musicinmybrain !

**Comment by fabiomt** [comment]
Uploaded 0.7.2 to Debian unstable, in a week should migrate to testing as well.

**Comment by scribblemaniac** [comment]
I have built and uploaded 32-bit Linux builds. I have changed the naming convention for them from i386 to i686 as that's likely more accurate, though I don't believe there have been any major changes in the build process since v0.7.0.

@chchwy Can you please make sure the v0.7.1 and v0.7.2 32-bit Linux binaries are added to the mirror downloads when you get a chance.

**Comment by chchwy** [comment]
> I have built and uploaded 32-bit Linux builds. I have changed the naming convention for them from i386 to i686 as that's likely more accurate, though I don't believe there have been any major changes in the build process since v0.7.0.
> 
> [@chchwy](https://github.com/chchwy) Can you please make sure the v0.7.1 and v0.7.2 32-bit Linux binaries are added to the mirror downloads when you get a chance.

Thank you @scribblemaniac. I've uploaded the linux 32bit binary to gitlab.
https://gitlab.com/chchwy/pencil2d/-/packages/56830084

**Comment by chchwy** [comment]
I am going to close this issue. Thank you everybody.
